Update to the HTTF lower plenum CFD modeling by adding nekRS model and documentation (#280)
idaholab/virtual_test_bed: doc/content/htgr/httf/lower_plenum_cfd.md

In the era of Exascale computing, there's a notable shift towards accelerated computing using Graphics Processing Units (GPUs) among leadership class supercomputers. To fully harness this new heterogeneous computing hardware, NekRS emerges as an exascale-ready thermal-fluids solver for incompressible and low-Mach number flows. Like its predecessor, Nek5000, it is based on the spectral element method and is scalable to $10^5$ processors and beyond. The principal differences between Nek5000 and NekRS are that the former is based on F77/C, coupled with MPI and fast matrix-matrix product routines for performance, while the latter is based on C++ coupled with MPI and OCCA kernels for GPU support. The Open Concurrent Compute Abstraction provides backends for CUDA, HIP, OpenCL, and DPC++, for performance portability across all the major GPU vendors. We note that, at 80% parallel efficiency, which corresponds to 2–3 million points per MPI rank, NekRS is running at 0.1 to 0.3 seconds of wall clock time per timestep, which is nearly a 3x improvement over production runs of Nek5000 at a similar 80% strong-scale limit on CPU-based platforms.

### Boundary conditions style=font-size:125%

> The boundary conditions of the nekRS study are taken from the corresponding system modeling of HTTF primary loop using RELAP5-3D conducted by Canadian Nuclear Laboratories [!citep](Podila2022). The reference lower plenum pressure is 211.9 kPa, and the helium gas has a density of 0.1950 kg/m3 with a reference temperature of 895.7 K. 

Reviewer: ```suggestion
The boundary conditions of the nekRS study are taken from the corresponding system modeling of HTTF primary loop using RELAP5-3D conducted by Canadian Nuclear Laboratories [!citep](Podila2022). The reference lower plenum pressure is 211.9 kPa, and the helium gas has a density of 0.1950 kg/m$^3$ with a reference temperature of 895.7 K. 
```

| CG-3E | 27.101 | 932.826 |
| CG-4E | 27.129 | 913.384 |
| CG-5E | 22.303 | 856.71 |
| CG-1F | 23.521 | 922.956 |
| CG-2F | 25.796 | 886.341 |
| CG-3F | 26.775 | 941.027 |
| CG-4F | 26.589 | 922.725 |
| CG-5F | 21.911 | 863.508 |
| Outer bypass | 20.669 | 810.051 |

!media httf/lower_plenum_cfd/new_inlet_bc.png
       style=width:75%
       id=new_bc
       caption=Boundary conditions: (a) prescribed inlet velocity and no-slip wall condition; (b) prescribed inlet temperature and adiabatic walls.

### NekRS case setups style=font-size:125%

Readers can access the nekRS case files and the corresponding mesh files in the VTB repository via GitHub Large File Storage (LFS). The mesh data is contained in two files: +httf.re2+ (which includes grid coordinates and sideset ids) and +httf.co2+ (containing mesh cell connectivity information).
Regarding the nekRS case files, there are four basic files:

- +httf.udf+ serves as the primary nekRS kernel file and encompasses algorithms executed on the hosts. It's responsible for configuring RANS model settings, collecting time-averaged statistics, and determining the frequency of calls to +userchk+, which is defined in the +usr+ file.

> - +httf.oudf+ is a supplementary file housing kernel functions for devices and also plays a role in specifying boundary conditions..

Reviewer: ```suggestion
- +httf.oudf+ is a supplementary file housing kernel functions for devices and also plays a role in specifying boundary conditions.
```

> - +httf.usr+ s a legacy file inherited from Nek5000 and can be utilized to establish initial conditions and define post-processing capabilities.

Reviewer: ```suggestion
- +httf.usr+ is a legacy file inherited from Nek5000 and can be utilized to establish initial conditions and define post-processing capabilities.
```
